### PR TITLE
docs: update example to use `date_time` type instead of `string`

### PR DESCRIPTION
**Pull request by sassela**
Addresses https://github.com/getsynth/synth/issues/257

Before:
```bash
➜  synth generate try-synth/ --size 2 | jq
Error: Unable to open the namespace "try-synth/"

Caused by:
    0: at file try-synth/users.json
    1: Failed to parse collection
    2: unknown variant `date_time`, expected one of `pattern`, `faker`, `categorical`, `serialized`, `uuid`, `truncated`, `format` at line 29 column 1
```

After:
```bash
➜  synth generate try-synth/ --size 2 | jq
{
  "users": [
    {
      "email": "margarete@example.org",
      "id": 1,
      "joined_on": "2016-03-04"
    },
    {
      "email": "brody@example.com",
      "id": 2,
      "joined_on": "2016-07-08"
    }
  ]
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,8 +102,9 @@ Next let's create a `users` collection using Synth's configuration language, and
             }
         },
         "joined_on": {
-            "type": "string",
-            "date_time": {
+            "type": "object",
```
Comment (WiredSound) [inline README.md:105]: Might make more sense to just have this as `"joined_on": { "type": "date_time", ... }` - i.e. remove the nested object.

File: README.md [hunk 2]
```diff
@@ -123,12 +124,16 @@ $ synth generate my_app/ --size 2 | jq
     {
       "email": "patricia40@jordan.com",
       "id": 1,
-      "joined_on": "2014-12-14"
+      "joined_on": {
```
Comment (WiredSound) [inline README.md:127]: These two other changes could then be changed back!

**Review by WiredSound — CHANGES_REQUESTED** [review]
Hey @sassela , thank you for your PR and sorry for the delay in getting back to you! Made some tiny (more stylistic rather than practical) suggestions. Thanks again

**Comment by sassela** [comment]
thanks @WiredSound ! Makes sense, I've applied your suggestion

**Pull request merged**